pixiefairy: init at 0.2.5

xhalo32 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This is my fork of the [pixiefairy](https://github.com/mbovo/pixiefairy) Python application, which I need for `pixiecore` API mode, so I packaged it with Nix. I hope that my changes are merged to the original repository, but I'm also ready to maintain this tool myself. I have created and tested a NixOS module for this package, and plan to get it merged after this.

As far as I know, the original project had only few users, but a similar project [waitron](https://github.com/ns1/waitron) has over 100 stars. Therefore it might be that this project is too small to nixpkgs. However, being small is one of its strengths as it integrates with the pixiecore API enabling its full functionality: to PXE boot different systems based on MAC address.

Resources:
- [NixOS discourse thread](https://discourse.nixos.org/t/packaging-pixiefairy/64307).
- [PR to original author's repository](https://github.com/mbovo/pixiefairy/pull/77)

I hope that `pixiecore` maintainers (@bbigras and @Mic92) could comment whether they think `pixiefairy` is a good addition to the NixOS ecosystem. I think they integrate really well and make PXE booting different systems with NixOS a breeze.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. --> 

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

xhalo32 committed:
pixiefairy: init at 0.2.5

Sigmanificient commented on pkgs/by-name/pi/pixiefairy/package.nix:
```diff
@@ -0,0 +1,42 @@
+  src = fetchFromGitHub {
+    owner = "xhalo32";
+    repo = "pixiefairy";
+    rev = "e5cd76af9d92dfb6db9adedd4dc58f9772785fe7";
```
```suggestion
   tag = "v${version}";
```
since you have tags in your fork

xhalo32 replied:
Updated

Sigmanificient reviewed (changes requested):
I think your are missing a runtime dependency:

xhalo32 commented on pkgs/by-name/pi/pixiefairy/package.nix:
```diff
@@ -0,0 +1,42 @@
+    urllib3
+    uvicorn
+  ];
+  meta = {
```
Good catch, I completely forgot the tests

xhalo32 commented:
@Sigmanificient thanks for your help and patience!

I removed the `pkg_resources` dependency from the tests (v0.2.5) in favor of `importlib.metadata` (see deprecation note https://setuptools.pypa.io/en/latest/pkg_resources.html) and added the `pytestCheckHook`.

The tests in `pixiefairy` are really bare and only check for smoke really. I hope it's not a blocking issue for this getting merged.

xhalo32 committed:
Merge branch 'NixOS:master' into master
